src/NuGet.Core/NuGet.ProjectModel/TargetFrameworkInformation.cs

```diff
@@ -143,9 +135,170 @@ public bool Equals(TargetFrameworkInformation other)
                    StringComparer.OrdinalIgnoreCase.Equals(TargetAlias, other.TargetAlias);
         }
 
-        public TargetFrameworkInformation Clone()
+        /// <summary>
+        /// Returns copy of this with specified versions defined in the Central package versions management file. 
+        /// </summary>
+        public TargetFrameworkInformation WithCentralPackageVersions(ImmutableDictionary<string, CentralPackageVersion> versions)
+        {
+            versions ??= EmptyCentralPackageVersions;
+
+            if (CentralPackageVersions == versions)
+            {
+                return this;
+            }
+
+            return new TargetFrameworkInformation(this)
+            {
+                CentralPackageVersions = versions
+            };
+        }
+
+        /// <summary>
+        /// Returns copy of this with specified framework references. 
+        /// </summary>
+        public TargetFrameworkInformation WithFrameworkReferences(ImmutableHashSet<FrameworkDependency> references)
```

Comment:
We could probably get rid of this as it's only used in the codepaths that are doing a package spec construciton